packages/backend-modules/access/lib/perks/accessWithRole.js

```diff
@@ -21,7 +23,17 @@ const give = async (campaign, grant, recipient, settings, t, pgdb) => {
 }
 
 const revoke = async (grant, recipient, settings, pgdb) => {
-  const isRoleRevoked = await removeRole(grant, recipient, pgdb, settings.role)
+  let isRoleRevoked = false
+  if (settings.role === memberRole) {
+    isRoleRevoked = await removeMemberRole(
+      grant,
+      recipient,
+      findByRecipient,
+      pgdb,
+    )
+  } else {
+    isRoleRevoked = await removeRole(grant, recipient, pgdb, settings.role)
+  }
```

Comment:
Style:  `let` vars are harder to keep track on who is mutating it. Using `const` vars and/or a more verbose pattern is usually easier to read.

```suggestion
  if (settings.role === memberRole) {
    const isRoleRevoked = await removeMemberRole(
      grant,
      recipient,
      findByRecipient,
      pgdb,
    )

    if (isRoleRevoked) {
      debug('revoke', {
        recipient: recipient.id,
        revokedRole: settings.role,
      })
      return {
        recipient: recipient.id,
        revokedRole: settings.role,
        eventLogExtend: `.${settings.role}`,
      }
    }
  }

  const isRoleRevoked = await removeRole(grant, recipient, pgdb, settings.role)
```

Reply:
I agree, but don't you think it leads to a lot of code duplication in this case?

Reply:
You could also do it like that, couldn't you?

```
const isRoleRevoked =
    settings.role === MEMBER_ROLE
      ? await removeMemberRole(grant, recipient, findByRecipient, pgdb)
      : await removeRole(grant, recipient, pgdb, settings.role)